fp-bindgen/src/types/type_ident.rs

```diff
@@ -1,32 +1,42 @@
 use crate::primitives::Primitive;
-use std::{
-    convert::{Infallible, TryFrom},
-    fmt::Display,
-    str::FromStr,
-};
+use std::{convert::TryFrom, fmt::Display, str::FromStr};
 use syn::{PathArguments, TypeParamBound, TypePath, TypeTuple};
 
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 #[non_exhaustive]
 pub struct TypeIdent {
     pub name: String,
     pub generic_args: Vec<(TypeIdent, Vec<String>)>,
+    pub array_len: usize,
```

Comment:
I think I might have preferred to wrap it in on an `Option` to clarify that it is set conditionally (instead of giving `0` the special meaning that it isn't an array).

Reply:
Yes, that's a good idea. I'll rename the field to `array`, too. If we need to extend it at some point with other array-specific data then it can fit in a tuple inside that `Option`.